### DISCUSSION TITLE
Can you add some feature to look at the log files

### DISCUSSION THREAD

**Discussion by XeonG** [oobabooga/text-generation-webui#2563] [Ideas]
Sometimes it would just be nice to have have some way of looking back at past chats with other models..

ie can you please save the chat/instruct logs etc with the actual llama model used.. and not the chat mode/model bleh.. I've noticed its not always saving chats with models for whatever reason, would just rather have all text and sent and received back logged under the actual llama model used instead ...and some ui feature to look at those so it looks properly formatted and not the json file.

**Comment by pleabargain** [comment]
I will one up your request. 
I want to be able to pass the responses back to the 'character' as a form of reinforcement learning.

I cannot find a way to see the entire logs of the conversations/output!

**Reply by luthis1124** [reply]
For reinforcement learning, you are better off using:

https://github.com/wawawario2/long_term_memory

There will be no learning except within the context window of 2000 tokens, so basically pointless. Perhaps you should look into training a LoRA if you want permanent 'memories'

**Reply by pleabargain** [reply]
Thanks for the tip!

**Comment by XeonG** [comment]
Or a maybe an extension that just saves the chat logs in a seperate folder named after the the loaded model?

**Reply by pleabargain** [reply]
An extension that needs to be  coded right? :) 
I see a world where apps will be able to teach users how to use the app because the app already has it's own LLM built in. What do you think?
